readthedocs/projects/views/private.py

```diff
@@ -161,25 +168,36 @@ def get_success_url(self):
         return reverse('projects_dashboard')
 
 
-@login_required
-def project_version_detail(request, project_slug, version_slug):
-    """Project version detail page."""
-    project = get_object_or_404(
-        Project.objects.for_admin_user(request.user),
-        slug=project_slug,
-    )
-    version = get_object_or_404(
-        Version.internal.public(
-            user=request.user,
-            project=project,
+class ProjectVersionMixin(ProjectAdminMixin, PrivateViewMixin):
+
+    model = Version
+    context_object_name = 'version'
+    form_class = VersionForm
+    lookup_url_kwarg = 'version_slug'
+    lookup_field = 'slug'
+
+    def get_success_url(self):
+        return reverse(
+            'project_version_list',
+            kwargs={'project_slug': self.get_project().slug},
+        )
+
+
+class ProjectVersionDetail(ProjectVersionMixin, UpdateView):
+
+    template_name = 'projects/project_version_detail.html'
+
+    def get_queryset(self):
+        return Version.internal.public(
+            user=self.request.user,
+            project=self.get_project(),
             only_active=False,
-        ),
-        slug=version_slug,
-    )
+        )
 
-    form = VersionForm(request.POST or None, instance=version)
+    def get_form(self, data=None, files=None, **kwargs):
+        return self.get_form_class()(data, files, **kwargs)
```

Comment:
Is this method needed? It does not seem to do anything different that the default.

Reply:
This overrides the one on `ProjectAdminMixin`, adding a comment